src/Ion.ts

```diff
@@ -41,47 +41,48 @@ function isBinary(buffer: Uint8Array): boolean {
   return true;
 }
 
+/** Octet buffer input types for the Ion reader interface. */
+export type ReaderOctetBuffer = ArrayBufferLike | ArrayLike<number>;
+
+/** All buffer input types for the Ion reader interface. */
+export type ReaderBuffer = ReaderOctetBuffer | string;
 
 /**
- * Create an Ion Reader object from a currentBuffer `buf`
+ * Create an {Reader} over Ion data in a {ReaderBuffer}.
  *
  * @param buf       The Ion data to be used by the reader. Typically a string, UTF-8 encoded buffer (text), or raw
  *                  binary buffer.
  * @param catalog   An optional {Catalog} to be used for resolving symbol table references.
  * @returns {Reader}
  */
-export function makeReader(buf: any, catalog? : Catalog) : Reader {
+export function makeReader(buf: ReaderBuffer, catalog? : Catalog) : Reader {
```

Comment:
I'm unaccustomed to `make` as a static factory method verb. I feel like `new` and `create` are more common. Thoughts?

Reply:
I don't feel super strong about the spelling here, personally. I'm good with whatever folks want here.

Reply:
whatevers good if were gonna change it we should do it now